### PR TITLE
Add info about cortex-tenant to the docs in Auth guide

### PR DESCRIPTION
**Pull request by blind-oracle**
**What this PR does**:
Adds a section in Auth guide that mentions cortex-tenant proxy that can be used to add tenant headers to Prometheus write requests based on labels.

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/guides/authentication-and-authorisation.md
Comment (pracucci) [inline docs/guides/authentication-and-authorisation.md:44]:
```suggestion
into distinct namespaces by some criteria (eg. teams, applications, etc).
```

File: docs/guides/authentication-and-authorisation.md
Comment (pracucci) [inline docs/guides/authentication-and-authorisation.md:38]: For transparency, I would just explicitly mention that `cortex-tenant` is a community project, not maintained by Cortex maintainers.

**Review by pracucci — APPROVED** [review]
Thanks! LGTM modulo a couple of nits.

File: docs/guides/authentication-and-authorisation.md
Comment (blind-oracle) [reply docs/guides/authentication-and-authorisation.md:38]: Yep, thought about this but wasn't sure how to say this properly. Done.

**Review by khaines — APPROVED** [review]

File: docs/guides/authentication-and-authorisation.md
Comment (pracucci) [inline docs/guides/authentication-and-authorisation.md:46]:
```suggestion
Be advised that **cortex-tenant** is a third-party community project and it's not maintained by Cortex team.
```

**Pull request merged**
